src/reconcile/rec_visibility.c

```diff
@@ -265,18 +265,6 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         if (upd->start_ts > max_ts)
             max_ts = upd->start_ts;
 
-        /*
-         * FIXME-prepare-support: A temporary solution for not storing durable timestamp in the
-         * cell. Properly fix this problem in PM-1524. It is currently not OK to write prepared
-         * updates with durable timestamp larger than checkpoint timestamp to data store as we don't
-         * store durable timestamp in the cell. However, it is OK to write them to the history store
-         * as we store the durable timestamp in the history store value.
-         */
-        if (upd->durable_ts != upd->start_ts && upd->durable_ts > checkpoint_timestamp) {
-            has_newer_updates = true;
-            continue;
-        }
-
         /* Always select the newest committed update to write to disk */
         if (upd_select->upd == NULL)
```

Comment:
There is a place we still set durable_ts to start_ts in this file. We should fix that.
```
/*
             * It's possible to have a tombstone as the only update in the update list. If we
             * reconciled before with only a single update and then read the page back into cache,
             * we'll have an empty update list. And applying a delete on top of that will result in
             * ONLY a tombstone in the update list.
             *
             * In this case, we should leave the selected update unset to indicate that we want to
             * keep the same on-disk value but set the stop time pair to indicate that the validity
             * window ends when this tombstone started.
             */
            upd_select->start_durable_ts = upd_select->start_ts = vpack->start_ts;
            upd_select->start_txn = vpack->start_txn;
```

Reply:
And I don't understand why we need to do these tests anymore:
```
            upd_select->stop_ts = upd->start_ts;
            if (upd->txnid != WT_TXN_NONE)
                upd_select->stop_txn = upd->txnid;
            if (upd->durable_ts != WT_TS_NONE)
                tombstone_durable_ts = upd->durable_ts;
```
Can we also remove `tombstone_durable_ts`?

Reply:
Just change it to:
```
            upd_select->stop_ts = upd->start_ts;
            upd_select->stop_txn = upd->txnid;
            upd_select->stop_durable_ts = upd->durable_ts;
```

Reply:
And this code section also:

```
            /* The beginning of the validity window is the selected update's time pair. */
            upd_select->start_durable_ts = upd_select->start_ts = upd->start_ts;
            /* If durable timestamp is provided, use it. */
            if (upd->durable_ts != WT_TS_NONE)
                upd_select->start_durable_ts = upd->durable_ts;
            upd_select->start_txn = upd->txnid;

            /* Use the tombstone durable timestamp as the overall durable timestamp if it exists. */
            if (tombstone_durable_ts != WT_TS_NONE)
                upd_select->stop_durable_ts = tombstone_durable_ts;
```

Reply:
Can we change it to
```
            /* The beginning of the validity window is the selected update's time pair. */
            upd_select->start_durable_ts = upd->durable_ts;
            upd_select->start_ts = upd->start_ts;
            upd_select->start_txn = upd->txnid;
```

Reply:
I reorganized the `__wt_rec_upd_select` function for the durable timestamps.